doc/source/flexibility/compressed_columns.rst

```diff
@@ -0,0 +1,290 @@
+through old read views are unsupported, similarly to |InnoDB| internal DDL
+transactions.
+
+Example
+-------
+
+In order to use the compression dictionary you'll need first to create it. This
+can be done by running:
+
+.. code-block:: mysql
+
+   mysql> SET @dictionary_data = 'one' 'two' 'three' 'four';
+   Query OK, 0 rows affected (0.00 sec)
+
+   mysql> CREATE COMPRESSION_DICTIONARY numbers (@dictionary_data);
+   Query OK, 0 rows affected (0.00 sec)
+
+To create a table that has both compression and compressed dictionary support
+you should run:
+
+.. code-block:: mysql
+
+   mysql> CREATE TABLE t1(
+           id INT,
+           a BLOB COLUMN_FORMAT COMPRESSED,
+           b BLOB COLUMN_FORMAT COMPRESSED WITH COMPRESSION_DICTIONARY numbers
+         ) ENGINE=InnoDB;
+
+Following example shows how to insert a sample of JSON data into the table:
+
+.. code-block:: mysql
+
+  SET @json_value =
+   '[\n'
+   ' {\n'
+   ' "one" = 0,\n'
+   ' "two" = 0,\n'
+   ' "three" = 0,\n'
+   ' "four" = 0\n'
+   ' },\n'
+   ' {\n'
+   ' "one" = 0,\n'
+   ' "two" = 0,\n'
+   ' "three" = 0,\n'
+   ' "four" = 0\n'
+   ' },\n'
+   ' {\n'
+   ' "one" = 0,\n'
+   ' "two" = 0,\n'
+   ' "three" = 0,\n'
+   ' "four" = 0\n'
+   ' },\n'
+   ' {\n'
+   ' "one" = 0,\n'
+   ' "two" = 0,\n'
+   ' "three" = 0,\n'
+   ' "four" = 0\n'
+   ' }\n'
+   ']\n'
+  ;
+
+.. code-block:: mysql
+
+  mysql> INSERT INTO t1 VALUES(0, @json_value, @json_value);
+  Query OK, 1 row affected (0.01 sec)
+
+
+INFORMATION_SCHEMA Tables
+=========================
+
+This feature implemented two new ``INFORMATION_SCHEMA`` tables.
+
+.. table:: INFORMATION_SCHEMA.XTRADB_ZIP_DICT
+
+  :column BIGINT(21)_UNSIGNED id: dictionary ID
+  :column VARCHAR(64) name: dictionary name
+  :column BLOB zip_dict: compressed dictionary
+
+This table stores the compression dictionary itself. ``SUPER`` privilege is
+required to query it.
+
+.. table:: INFORMATION_SCHEMA.XTRADB_ZIP_DICT_COLS
+
+  :column BIGINT(21)_UNSIGNED table_id: table ID from ``INFORMATION_SCHEMA.INNODB_SYS_TABLES``
+  :column BIGINT(21)_UNSIGNED column_pos: column position (starts from ``0`` as in ``INFORMATION_SCHEMA.INNODB_SYS_COLUMNS``)
+  :column BIGINT(21)_UNSIGNED dict_id: dictionary ID
+
+This table stores the mapping between the compression dictionary and columns
+using the dictionary. ``SUPER`` privilege is require to query it.
+
+Limitations
+===========
+
+Compressed columns cannot be used in indices (neither on their own nor as parts
+of composite keys).
+
+.. note::
+
+  ``CREATE TABLE t2 AS SELECT * FROM t1`` will create a new table with a
+  compressed column, whereas ``CREATE TABLE t2 AS SELECT CONCAT(a,'') AS a FROM
+  t1`` will not create compressed columns.
+
+  At the same time, after executing ``CREATE TABLE t2 LIKE t1`` statement,
+  ``t2.a`` will have ``COMPRESSED`` attribute.
+
+``ALTER TABLE ... DISCARD/IMPORT TABLESPACE`` is not supported for tables with
+compressed columns.
```

Comment:
To export and import tablespaces with compressed columns, uncompress them first: "alter table column_format default example here"